sdks/python/apache_beam/ml/inference/pytorch_inference.py

```diff
@@ -201,20 +207,30 @@ def __init__(
       logging.info("Device is set to CPU")
       self._device = torch.device('cpu')
     self._model_class = model_class
-    self._model_params = model_params
+    self._model_params = model_params if model_params else {}
     self._inference_fn = inference_fn
-    self._use_torch_script_format = use_torch_script_format
+    self._batching_kwargs = {}
+    if min_batch_size is not None:
+      self._batching_kwargs['min_batch_size'] = min_batch_size
+    if max_batch_size is not None:
+      self._batching_kwargs['max_batch_size'] = max_batch_size
+    self._torch_script_model_path = torch_script_model_path
 
-    self._validate_func_args()
+    self.validate_constructor_args()
 
-  def _validate_func_args(self):
-    if not self._use_torch_script_format and (self._model_class is None or
-                                              self._model_params is None):
+  def validate_constructor_args(self):
+    if self._state_dict_path and not self._model_class:
```

Comment:
```suggestion
    if bool(self._state_dict_path) != bool(self._model_class):
```

or alternatively:

```suggestion
    if (self._state_dict_path and not self._model_class) or (not self._state_dict_path and self._model_class):
```

We just need to catch the case where they provide the model class but not the state_dict_path